### DISCUSSION TITLE
Parallel simulation error

### DISCUSSION THREAD

**Discussion by nahidrahmati** [UvaCsl/HemoCell#50] [General discussion]
Hi,
I want to know in which situations, the cells are removing? I know one of them is related to overlapping or being close to the wall. In my case, when I run parallel the case, the cell are removing even when they are in the middle of the domain. but, when I run it on single core, they are not removing. Does parallel run affect of deleting the Cell? 

Other question is related to how many core should we use for running the simulation. because before starting the hemocell.iterate, there is a warning related to the block size. Should the number of core we are using, be in a way to meet the block size condition?

**Comment by nahidrahmati** [comment]
It is worth mentioning that, when the Cells  are removing from the domain, I will not see any issue warning.

**Comment by gzavo** [comment]
This is not enough information to say anything conclusive. Cells are removed at the beginning of a simulation (without warning) only if they are outside of the simulation domain, or outside of the fluid domain, or intersect with a boundary. If they are present initially, but removed after some time-steps, then there is a warning message. There are examples (oneCEllShear and cellStretch) that are meant for a single RBC, hence they are not parallel. In every other example parallel run should not be an issue.

**Comment by mojtabaAmiraslanpour** [comment]
I want to report the same issue here. I have tested this with two various _Number of atomic blocks_:

1st simulation
=========
```
Size of the multi-block:     641-by-159-by-42
Number of atomic-blocks:     96
Smallest atomic-block:       106-by-39-by-10
Largest atomic-block:        107-by-40-by-11
Number of allocated cells:   4.2806 million
Fraction of allocated domain: 100 percent
...
(main) Stats. @ 1000 (0.0001 s):
	 # of cells: 412 | # of RBC: 375, PLT: 37
...
(main) Stats. @ 750000 (0.075 s):
	 # of cells: 127 | # of RBC: 95, PLT: 32
```

2nd simulation
=========
```
Size of the multi-block:     641-by-159-by-42
Number of atomic-blocks:     48
Smallest atomic-block:       106-by-39-by-21
Largest atomic-block:        107-by-40-by-21
Number of allocated cells:   4.2806 million
Fraction of allocated domain: 100 percent
...
(main) Stats. @ 1000 (0.0001 s):
	 # of cells: 411 | # of RBC: 374, PLT: 37
...
(main) Stats. @ 750000 (0.075 s):
	 # of cells: 407 | # of RBC: 370, PLT: 37
```

As you can see, for the same problem, the number of RBCs decreases significantly after using a greater number of atomic blocks for parallel simulation. FYI I do not get any warnings in the middle of the run either.

**Reply by nahidrahmati** [reply]
Hi Mojtaba, 
Since you had the RBCs in the initial time steps, and they are removing after some time steps, it seems that RBCs are intersecting with the boundary domain. Did you activate the repulsion force and increase its constant in a way to avoid removing the cells?

**Reply by mojtabaAmiraslanpour** [reply]
Hi Nahid.
Thank you for the hint. I have repulsive force enabled but I have set it to be small not to impact the overall physics of the flow. Have you been able to remove this issue by increasing the force further?

**Reply by mojtabaAmiraslanpour** [reply]
To be honest, I do not know why these two can be related to each other. Why the number of atomic blocks has anything to do with the cell deletion? This should be more about atomic blocks boundaries than the wall boundary itself.

**Comment by gzavo** [comment]
Hi,

Have you turned on the information on deleted cells?

See the relevant documentation here:
https://github.com/UvaCsl/HemoCell/blob/3a4c777821bc90a0292368de8dbdbd332eadb2e0/doc/user_guide/xml_files.rst?plain=1#L23-L25

You can set it like this:
https://github.com/UvaCsl/HemoCell/blob/ccccc63aacc211847e79f4478cf701e78663487b/examples/parallelplanes/config.xml#L5

**Reply by mojtabaAmiraslanpour** [reply]
I see. No, I forgot all about that! Will try it.

**Reply by mojtabaAmiraslanpour** [reply]
Well, this gives me the location of the deleted cells now. Deleted cells are outside of the domain. But how can I fix this?
